usr/src/cmd/zpool/zpool_main.c

```diff
@@ -617,6 +623,139 @@ zpool_do_remove(int argc, char **argv)
 }
 
 /*
+ * zpool labelclear <vdev>
+ *
+ * Verifies that the vdev is not active and zeros out the label information
+ * on the device.
+ */
+int
+zpool_do_labelclear(int argc, char **argv)
+{
+	char *vdev, *name = NULL;
+	struct stat st;
+	int c, fd = -1, ret = 0;
+	pool_state_t state;
+	boolean_t inuse = B_FALSE;
+	boolean_t force = B_FALSE;
+
+	/* check options */
+	while ((c = getopt(argc, argv, "f")) != -1) {
+		switch (c) {
+		case 'f':
+			force = B_TRUE;
+			break;
+		default:
+			(void) fprintf(stderr, gettext("invalid option '%c'\n"),
+			    optopt);
+			usage(B_FALSE);
+		}
+	}
+
+	argc -= optind;
+	argv += optind;
+
+	/* get vdev name */
+	if (argc < 1) {
+		(void) fprintf(stderr, gettext("missing vdev device name\n"));
+		usage(B_FALSE);
+	}
+
+	/* Allow bare paths if they exist, otherwise prepend. */
+	if (stat(argv[0], &st) != 0 &&
+	    strncmp(argv[0], ZFS_DISK_ROOTD, strlen(ZFS_DISK_ROOTD)) != 0)
+		ret = asprintf(&vdev, ZFS_DISK_ROOTD "%s", argv[0]);
+	else
+		ret = asprintf(&vdev, "%s", argv[0]);
+
+	if (ret < 0) {
+		perror("asprintf");
+		return (ret);
+	}
+
+	if ((fd = open(vdev, O_RDWR)) < 0) {
+		ret = errno;
+		(void) fprintf(stderr,
+		    gettext("Unable to open %s: %s\n"), vdev, strerror(errno));
+		goto errout;
+	}
+
+	ret = zpool_in_use(g_zfs, fd, &state, &name, &inuse);
+	if (ret != 0) {
+		if (force)
+			goto wipe_label;
+
+		(void) fprintf(stderr,
+		    gettext("Unable to determine pool state for %s\n"
+		    "Use -f to force the clearing any label data\n"), vdev);
+		goto errout;
+	}
+
+	if (inuse) {
+		switch (state) {
+		default:
+		case POOL_STATE_ACTIVE:
+		case POOL_STATE_SPARE:
+		case POOL_STATE_L2CACHE:
+			(void) fprintf(stderr,
+gettext("labelclear operation failed.\n"
```

Comment:
This is from the original version.  The next best option I can think of would be to create preprocessor constants that contain the format strings used here, and use them in place.  Would that be OK; any better suggestions?